src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -511,23 +511,23 @@ private void writeRecord(SinkRecord record) throws IOException {
     long expectedOffset = offset + recordCounter;
     if (offset == -1) {
       offset = record.kafkaOffset();
-    } else if (record.kafkaOffset() != expectedOffset) {
+    } else if (record.kafkaOffset() < expectedOffset) {
```

Comment:
Hmm, 2 things. First, I think if we change this then we want `expectedOffset` to be computed differently. Right now it just adds the record count, but that wouldn't make sense anymore.

Secondly, maybe we can just remove this entirely? Or just move the check behind a config flag maybe so there's a way to turn it off if you don't need it? The underlying issue that required this check is fixed in 0.10.1.0, although removing this entirely would put folks trying to run a newer version of the connector in an older cluster at risk.